psi4/driver/procrouting/diis.py

```diff
@@ -181,33 +232,170 @@ def extrapolate(self, *args):
         if np.all(diagonals > 0):
             diagonals = diagonals ** (- 0.5)
             B = np.einsum("i,ij,j -> ij", diagonals, B, diagonals)
-            coeffs = np.linalg.lstsq(B, rhs, rcond=None)[0][:-1] * diagonals[:-1]
+            return np.linalg.lstsq(B, rhs, rcond=None)[0][:-1] * diagonals[:-1]
         else:
-            coeffs = np.linalg.lstsq(B, rhs, rcond=None)[0][:-1]
+            return np.linalg.lstsq(B, rhs, rcond=None)[0][:-1]
+
+    def adiis_energy(self, x):
+        x = normalize_input(x)
+        return np.dot(self.adiis_linear, x) + np.einsum("i,ij,j->", x, self.adiis_quadratic, x) / 2
+
+    def adiis_gradient(self, x):
```

Comment:
Docstring might be good to add even though function name is pretty self-explanatory